### ISSUE TITLE
Installation of mne_icalabel

### ISSUE THREAD

**Issue by mscheltienne** [jacobf18/iclabel-python#6]
Hello all, I am trying to install/import iclabel-mne but not successful. For a starting purpose I will try to execute the above example by @adam2392 but I am not sure what I am doing wrong. I tried to install mne-iclabel via pip install method mentioned in [here](https://github.com/jacobf18/iclabel-python) via following

(NOTE: i am working in colab that's why exclamation mark before pip command)
`!pip install --user -U https://api.github.com/repos/jacobf18/iclabel-python/zipball/main`

and get the message that mne-icalabel-0.1.dev0 is successfully installed, see below

`Collecting https://api.github.com/repos/jacobf18/iclabel-python/zipball/main`
`  Downloading https://api.github.com/repos/jacobf18/iclabel-python/zipball/main`
`     - 122.7 MB 1.2 MB/s`
`Collecting mne>=0.24`
`  Downloading mne-0.24.1-py3-none-any.whl (7.4 MB)`
`     |████████████████████████████████| 7.4 MB 6.0 MB/s `
`Requirement already satisfied: numpy>=1.16.0 in /usr/local/lib/python3.7/dist-packages (from mne-icalabel==0.1.dev0) (1.21.5)`
`Requirement already satisfied: scipy>=1.2.0 in /usr/local/lib/python3.7/dist-packages (from mne-icalabel==0.1.dev0) (1.4.1)`
`Requirement already satisfied: setuptools in /usr/local/lib/python3.7/dist-packages (from mne-icalabel==0.1.dev0) (57.4.0)`
`Building wheels for collected packages: mne-icalabel`
`  Building wheel for mne-icalabel (setup.py) ... done`
`  Created wheel for mne-icalabel: filename=mne_icalabel-0.1.dev0-py2.py3-none-any.whl size=17612 sha256=6e077310f31de483cd83b50dcf6f88df93ba8f12a45b8cee05938a163e46a054`
`  Stored in directory: /tmp/pip-ephem-wheel-cache-be7s5va2/wheels/d4/00/98/7e59c6dea60d544087ce3d8d4067578061db5645b5538a1890`
`Successfully built mne-icalabel`
`Installing collected packages: mne, mne-icalabel`
`  WARNING: The script mne is installed in '/root/.local/bin' which is not on PATH.`
`  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.`
`Successfully installed mne-0.24.1 mne-icalabel-0.1.dev0`

but when I try to import the modules it says "ModuleNotFoundError: No module named 'mne_icalabel'". see below

`# import`
`from mne_icalabel.ica_label import ica_eeg_features`
`from mne_icalabel.ica_net import run_iclabel`

`ModuleNotFoundError                       Traceback (most recent call last)`
`<ipython-input-2-a755f923046f> in <module>()`
`      # import`
`----> from mne_icalabel.ica_label import ica_eeg_features`
`      from mne_icalabel.ica_net import run_iclabel`
`ModuleNotFoundError: No module named 'mne_icalabel'`
`---------------------------------------------------------------------------`
`NOTE: If your import is failing due to a missing package, you can`
`manually install dependencies using either !pip or !apt.`
`To view examples of installing some common dependencies, click the`
`"Open Examples" button below.`
`---------------------------------------------------------------------------`

Could you please let me know if my installation is not correct? Should there be another way to install mne-iclable? if yes, could you please update its procedure in the [main repo](https://github.com/jacobf18/iclabel-python) for future users.

any help would be much appreciated.
thanks
Rab

_Originally posted by @rnawaz in https://github.com/jacobf18/iclabel-python/issues/5#issuecomment-1070797656_

**Comment by rnawaz** [comment]
Thanks @mscheltienne for re-posting and opening a new issue. But, could you please link [#5 (comment)](https://github.com/jacobf18/iclabel-python/issues/5#issue-1159925942) to "**above example**" in the second sentence of my post. Just to avoid any mismatching of the thoughts.

**Comment by mscheltienne** [comment]
@rnawaz Your problem is not related to the import lines. You copy/pasted the code from the other issue and it is correct.

The problem is that despite the installation, the package is not present in your environment.
As I am not familiar with colab, I don't think I can help you much with this part.

Also, this package is still under development and has not yet been fully tested/verified against the Matlab implementation, so use it at your own risk.

**Comment by rnawaz** [comment]
@mscheltienne Colab is just a cloud IDE for python. 
Anyway, I tried to install the mne-iclabel in my local python setup via the following command

`pip install --user -U https://api.github.com/repos/jacobf18/iclabel-python/zipball/main`

and again the installation was successful. But when I tried to import using the following lines

`from mne_icalabel.ica_label import ica_eeg_features`
`from mne_icalabel.ica_net import run_iclabel`

I get the same module error saying "ModuleNotFoundError: No module named 'mne_icalabel'".

This time I am using Jupyter notebook and not Colab.

**Comment by mscheltienne** [comment]
Yes, I know what Colab is, and I thought the cloud-aspect might have caused the problem.
Clone the repository and then run `python setup.py install` or `python setup.py develop`, or with `pip`:

```
pip install git+https://github.com/jacobf18/iclabel-python
```

![Screenshot 2022-03-17 at 14 39 34](https://user-images.githubusercontent.com/73893616/158820811-57a343d0-18e2-4b11-9b17-8e5709ea3d98.png)

P.S: also for code formatting on GitHub, you can use triple backticks ` before and after a code block to format multiple lines.

**Comment by mscheltienne** [comment]
That is a surprising one.. I'm even surprised `pip install` worked on a Windows command prompt. On windows I use `py -m pip install` instead. I'm guessing you have multiple installs of pythons (environments) and the command `python` points to this 3.8.2 install while the command `pip` points to another one. 

Probably if you do `python -m pip install git+https://github.com/jacobf18/iclabel-python` that will work (since this time you call the command python so you should be in the same env.)

**Comment by adam2392** [comment]
I cannot replicate this installation issue, so not sure how to help :/. 

But also this repo is in super-alpha mode, so there is no guarantee of anything yet.

Once we establish an explicit understanding of how this repo differs from EEGLab's implementation of ICLabel, then we will thoroughly document and possibly move this repo under mne.tools GitHub organization.

**Comment by mscheltienne** [comment]
Great. So I'll close this one, as the install part is solved.

But don't expect much support on the use-part for now, as Adam pointed out, this is a very early version, not verified, not tested, not documented, and not ready for use. You can play with it as much as you want, but you'll have to figure out on your own how it works, and there is no guarantee that the results outputted by the network are correct (at least for now).
